### PR TITLE
Simplify strategy installation in CI

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Since some strategies are now external, we can simplify the CI setup

cc @carmocca @borda

### CODE REVIEW

File: .azure/gpu-benchmark.yml [hunk 1]
```diff
@@ -76,7 +75,6 @@ jobs:
 
     - bash: |
         set -e
-        pip list
```
Comment (carmocca) [inline .azure/gpu-benchmark.yml:79]: collect_env_details already includes pip list

File: .azure/gpu-benchmark.yml [hunk 2]
```diff
@@ -59,7 +59,6 @@ jobs:
     - bash: |
         echo $CUDA_VISIBLE_DEVICES
         echo $TORCH_URL
-        lspci | egrep 'VGA|3D'
```
Comment (carmocca) [inline .azure/gpu-benchmark.yml:62]: lspci is not available

File: requirements/pytorch/check-avail-strategies.py [hunk 3]
```diff
@@ -1,2 +0,0 @@
-if __name__ == "__main__":
-    import deepspeed  # noqa: F401
```
Comment (awaelchli) [inline requirements/pytorch/check-avail-strategies.py:2]: this looks like it was here on purpose.

File: .azure/gpu-tests-pytorch.yml [hunk 4]
```diff
@@ -51,10 +51,6 @@ jobs:
     cancelTimeoutInMinutes: "2"
     strategy:
       matrix:
-        'PyTorch & strategies':  # this uses torch 1.12 as not all strategies support 1.13 yet
```
Comment (awaelchli) [inline .azure/gpu-tests-pytorch.yml:54]: why remove this from the matrix?
we should still test with deepspeed, it is not external

Comment (Borda) [reply .azure/gpu-tests-pytorch.yml:54]: in such case we need to add `deepspeed` to all runs

File: .azure/gpu-tests-pytorch.yml [hunk 5]
```diff
@@ -106,29 +101,21 @@ jobs:
         for fpath in `ls requirements/**/*.txt`; do \
           python ./requirements/pytorch/adjust-versions.py $fpath ${PYTORCH_VERSION}; \
         done
+        # prune packages with installation issues
+        pip install -q -r .actions/requirements.txt
+        python .actions/assistant.py requirements_prune_pkgs \
```
Comment (Borda) [inline .azure/gpu-tests-pytorch.yml:106]: these shall not be listed among strategies as they introduce cyclic depedency

File: requirements/pytorch/check-avail-strategies.py (hunk 3 above)
Comment (Borda) [reply requirements/pytorch/check-avail-strategies.py:2]: agree

File: .azure/gpu-tests-pytorch.yml (hunk 4 above)
Comment (carmocca) [reply .azure/gpu-tests-pytorch.yml:54]: Adding deepspeed to all runs is what I did

File: .azure/gpu-tests-pytorch.yml (hunk 5 above)
Comment (carmocca) [reply .azure/gpu-tests-pytorch.yml:106]: They are not listed as strategies in setup or anything like that

File: requirements/pytorch/check-avail-strategies.py (hunk 3 above)
Comment (carmocca) [reply requirements/pytorch/check-avail-strategies.py:2]: It used to be useful when we had multiple external strategies in the repo. It's not the case anymore.

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**